scripts/apps/desks/controllers/DeskConfigController.ts

```diff
@@ -85,26 +87,46 @@ export function DeskConfigController($scope, $controller, notify, desks, modal)
         $scope.desk.edit = null;
     };
 
-    $scope.confirmSave = function() {
+    $scope.confirmSave = function(diff) {
         return modal.confirm(
             gettext('You have unsaved changes. Do you want to save them now?'),
             gettext('Save changes?'),
-            gettext('Yes'),
-            gettext('No'));
+            gettext('Save'),
+            gettext('Ignore'))
+            .then(() => {
+                return desks.save($scope.desk.orig, diff).then((res) => {
+                    _.merge($scope.desk.edit, res);
+                    _.merge($scope.desk.orig, res);
+                    if (diff.members) {
+                        // if desk members were changed update them inside desks.deskMembers as well
```

Comment:
You shouldn't change data owned by a service, outside of the service. If you need to update members, move it to the service.